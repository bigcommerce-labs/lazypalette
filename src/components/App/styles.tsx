import styled from 'styled-components';

export const StyledApp = styled.div`
    background: ${({ theme: { colors } }) => colors.background};
    align-content: flex-start;
    display: flex;
    flex-direction: column;
    height: 100%;
    position: fixed;
    width: 100%;
`;

export const Viewport = styled.div`
    display: flex;
    flex-direction: row;
    flex-grow: 1;
`;
