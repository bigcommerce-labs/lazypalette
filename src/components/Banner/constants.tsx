import React from 'react';

export const CopyMessage = 'Link copied to clipboard.';

export const TooltipMessage =
    <>
        <p>You must launch your store from the Getting Started page in order to make it public.</p>
        <p><a href="/manage/getting-started">Go to Getting Started</a></p>
        <p>You can still let other people see your store before it is launched by sharing the private store link.</p>
    </>;

export const MaintenanceMessage = 'Your store is down for maintenance.';
export const PrelaunchMessage = 'Your store has not launched and is not publicly visible.';
