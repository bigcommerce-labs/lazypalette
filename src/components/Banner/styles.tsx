import styled from 'styled-components';

export const StyledBanner = styled.div`
    background-color: #FFD24D;
    text-align: center;
    width: 100%;
    line-height: 2.5rem;
    font-size: 1rem;
    font-weight: 600;
    color: #303540;
`;

export const StyledIcon = styled.span`
    span {
        margin-bottom: -5px;
    }
`;

export const StyledMessage = styled.p`
    text-align: left;

    a {
        color: ${({ theme }) => theme.colors.primary};
        display: block;
        font-weight: bold;
        margin-top: 5px;
        text-decoration: none;
    }
`;

export const StyledShareableLink = styled.button`
    background: ${({ theme }) => theme.colors.warning};
    color: ${({ theme }) => theme.colors.primaryText};
    font-size: ${({ theme }) => theme.typography.fontSize.small};
    font-weight: ${({ theme }) => theme.typography.fontWeight.bold}
    font-family: ${({ theme }) => theme.typography.fontFamily.headings};
    margin: .5rem 0 .5rem 1.25rem;
    outline: none;
    border: 1px solid;
    border-radius: 2px;
    display: inline-block;
    padding: .25rem .5rem;
    line-height: 1.375rem;
    cursor: pointer;
`;
