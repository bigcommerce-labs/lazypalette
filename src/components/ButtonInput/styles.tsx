import styled from 'styled-components';

export const Wrapper = styled.div`
    margin: 0;
`;

interface ButtonProps {
    border?: boolean;
    classType?: string;
}

export const Button = styled.button.attrs<ButtonProps>({})`
    height: 2rem;
    border-radius: 2px;
    box-sizing: border-box;
    border: ${({ border, theme }) => border ? `1px solid ${theme.colors.primary}` : 'none'};
    padding: ${({ border }) => border ? '0 1rem' : '0'};
    font-family: ${({ theme }) => theme.typography.fontFamily.sans };
    font-size: 1rem;
    cursor: pointer;

    ${({ border, classType, theme }) => {
        const isPrimary = classType === 'primary';

        return `
            background-color: ${isPrimary ? theme.colors.primary : '#FFF'};
            color: ${isPrimary ? '#FFF' : theme.colors.primary};

            :hover {
                background-color: ${isPrimary ? theme.colors.primaryHover : theme.colors.secondaryHover};
                ${!border && `
                    background-color: transparent;
                `}
                border-color: ${isPrimary ? theme.colors.brandPrimary : theme.colors.primary};
                text-decoration: ${border ? 'none' : 'underline'};
            }

            :focus {
                outline: none !important;
            }

            :disabled {
                background-color: ${isPrimary ? theme.colors.primary : '#FFF'};
                opacity: 0.5;
            }
        `;
    }}
`;

Button.defaultProps = {
    theme: {
        colors: {
            background: '#F5F7FA',
            brandPrimary: '#273A8A',
            primary: '#3F69FE',
        },
        typography: {
            fontFamily: {
                sans: '"Source Sans Pro", "Helvetica Neue", Arial, sans-serif',
            },
        },
    },
};
