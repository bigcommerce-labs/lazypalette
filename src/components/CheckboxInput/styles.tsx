import styled from 'styled-components';

export const Container = styled.div`
    position: relative;
`;

export const Input = styled.input.attrs({
    type: 'checkbox',
})`
    ${({ theme }) => `
        opacity: 0;

        :not(:checked) + label:after {
            opacity: 0;
        }

        :not(:checked) + label {
            background: #FFF;
            border: 1px solid ${theme.colors.guideText};
            box-sizing: border-box;
        }
    `}
`;

export const Label = styled.label`
    ${({ theme }) => `
        cursor: pointer;
        position: absolute;
        width: 1.25rem;
        height: 1.25rem;
        top: 25%;
        left: 0;
        background: ${theme.colors.primary};
        border-radius: 2px;

        :after {
            content: '';
            position: absolute;
            width: .5rem;
            height: .25rem;
            background: transparent;
            top: .25rem;
            left: .25rem;
            border: 3px solid ${theme.colors.stroke};
            border-top: none;
            border-right: none;
            transform: rotate(-45deg);
        }
    `}
`;

export const HiddenLabel = styled.label`
    padding-left: .5rem;
    line-height: 1.25rem;
    display: inline-block;
`;
