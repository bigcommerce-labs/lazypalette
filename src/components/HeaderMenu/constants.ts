export const enum HeaderMenuLinks {
    ControlPanel = 'manage',
}

export const HeaderMenuMessages = {
    GoBack: 'Go back to Control Panel',
    ModalAction: 'Leave this page',
    ModalTitle: 'Leave Page?',
    UnsavedModalBody: `You have unsaved changes that will be lost if you decide to continue.
        Are you sure you want to leave this page?`,
};
