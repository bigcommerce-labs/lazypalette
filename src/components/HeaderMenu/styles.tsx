import { theme, BCPrimaryLogo, Icon } from 'pattern-lab';
import React from 'react';
import styled from 'styled-components';

import { ViewportType } from '../PreviewPane/PreviewPane';
import Tooltip from '../Tooltip/Tooltip';

export const StyledHeaderMenu = styled.nav`
    align-items: center;
    background: ${({ theme: { colors } }) => colors.empty};
    border-bottom: 1px solid ${({ theme: { colors } }) => colors.stroke};
    box-sizing: border-box;
    display: flex;
    flex-basis: 4rem;
    flex-shrink: 0;
    padding: 0 1.5rem 0 1.5rem;
`;

export const BCLogoInline = styled.div`
    cursor: pointer;
    display: inline-block;
`;

interface BCLogoProps {
    tooltip: string;
    onClick(): void;
}

export const BCLogo = (props: BCLogoProps) => (
    <Tooltip message={props.tooltip} >
        <BCLogoInline onClick={props.onClick}>
            <BCPrimaryLogo />
        </BCLogoInline>
    </Tooltip>
);

interface StyledHeaderSpanProps {
    isSelected: boolean;
}

export const StyledHeaderSpan = styled.span.attrs<StyledHeaderSpanProps>({})`
    background-color: ${({ isSelected, theme: { colors } }) => isSelected && colors.selectedBackground};
    cursor: pointer;
    display: inline-block;
    height: 2rem;
    margin-right: 0.5rem;

    :first-of-type {
        margin-left: 3.5rem
    }
`;

interface StyledPreviewItemProps {
    isRotated?: boolean;
    isSelected: boolean;
    tooltip: string;
    viewportType: ViewportType;
    onClick(): void;
}

interface StyledIconProps {
    isRotated?: boolean;
}

const StyledIcon = styled.div.attrs<StyledIconProps>({
    style: ({ isRotated }: StyledIconProps) => ({
        transform: isRotated ? 'rotate(-90deg) translate(2px,1px)' : 'none',
    }),
})`
    transform: ${(props: StyledIconProps) => props.isRotated ? 'rotate(-90deg) translate(2px,1px)' : 'none'};
    transition-duration: 0.5s;
`;

export const StyledPreviewItem = (props: StyledPreviewItemProps) => {
    return (

        <Tooltip
            message={props.tooltip}
        >
            <StyledHeaderSpan isSelected={props.isSelected} onClick={props.onClick}>
                <StyledIcon isRotated={props.isRotated || false} >
                    <Icon
                        glyph={props.viewportType.glyphName}
                        primaryColor={props.isSelected ? theme.colors.primaryText : theme.colors.primary}
                        size={'largest'}
                        data-test-id={props.viewportType.glyphName}
                    />
                </StyledIcon>
            </StyledHeaderSpan>
        </Tooltip>
    );
};
