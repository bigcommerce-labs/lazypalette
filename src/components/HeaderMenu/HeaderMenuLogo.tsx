import React, {PureComponent} from 'react';

import ConfirmModal from '../Modal/ConfirmModal/ConfirmModal';

import { HeaderMenuLinks, HeaderMenuMessages } from './constants';
import { BCLogo } from './styles';

interface HeaderMenuLogoProps {
    isChanged?: boolean;
}

interface HeaderMenuLogoState {
    isUnsavedOpen: boolean;
}

class HeaderMenuLogo extends PureComponent<HeaderMenuLogoProps> {
    readonly state: HeaderMenuLogoState = {
        isUnsavedOpen: false,
    };

    handleLogoClick = () => {
        const { isChanged } = this.props;

        if (isChanged) {
            this.setState({ isUnsavedOpen: true });
        } else {
            this.handleLogoLink();
        }
    };

    handleLogoLink = () => window.location.assign(`/${HeaderMenuLinks.ControlPanel}`);

    handleModalCancel = () => this.setState({ isUnsavedOpen: false });

    render() {
        const { isUnsavedOpen } = this.state;

        return (
            <>
                <BCLogo tooltip={HeaderMenuMessages.GoBack} onClick={this.handleLogoClick} />
                {isUnsavedOpen &&
                    <ConfirmModal
                        primaryAction={this.handleLogoLink}
                        primaryActionText={HeaderMenuMessages.ModalAction}
                        secondaryAction={this.handleModalCancel}
                        overlayClose={this.handleModalCancel}
                        title={HeaderMenuMessages.ModalTitle}
                    >
                        {HeaderMenuMessages.UnsavedModalBody}
                    </ConfirmModal>
                }
            </>
        );
    }
}

export default HeaderMenuLogo;
