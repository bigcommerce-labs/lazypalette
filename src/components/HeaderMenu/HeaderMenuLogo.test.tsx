import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';

import ConfirmModal from '../Modal/ConfirmModal/ConfirmModal';

import { HeaderMenuLinks } from './constants';
import { BCLogoInline } from './styles';
import HeaderMenuLogo from './HeaderMenuLogo';

describe('HeaderMenuLogo', () => {
    describe('when rendering', () => {
        it('renders the component', () => {
            const headerMenu = shallow(<HeaderMenuLogo isChanged={false} />);

            expect(toJson(headerMenu)).toMatchSnapshot();
        });
    });

    describe('when handleLogo is clicked', () => {
        describe('when there are no theme changes', () => {
            it('invokes window.location.assign with control panel address', () => {
                const component = mount(
                    <HeaderMenuLogo isChanged={false} />
                );

                const _window: any = window;
                _window.location.assign = jest.fn();

                expect(_window.location.assign).not.toBeCalled();
                component.find(BCLogoInline).simulate('click');
                expect(_window.location.assign).toBeCalledWith(
                    `/${HeaderMenuLinks.ControlPanel}`
                );
            });
        });

        describe('when there are unsaved theme changes', () => {
            it('should open up the confirm dialog modal', () => {
                const component = mount(
                    <HeaderMenuLogo isChanged={true} />
                );

                expect(component.find(ConfirmModal).exists()).toEqual(false);
                component.find(BCLogoInline).simulate('click');
                expect(component.find(ConfirmModal).exists()).toEqual(true);
            });
        });
    });
});
