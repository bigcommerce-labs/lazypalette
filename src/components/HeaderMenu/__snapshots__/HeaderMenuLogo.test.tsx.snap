// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderMenuLogo when rendering renders the component 1`] = `
<Fragment>
  <Component
    onClick={[Function]}
    tooltip="Go back to Control Panel"
  />
</Fragment>
`;
